### PR TITLE
[ggj] feat: add streaming parsing and proto test starters

### PR DESCRIPTION
**Pull request by miraleung**

### CODE REVIEW

File: src/main/java/com/google/api/generator/gapic/model/Method.java [hunk 1]
```diff
@@ -19,16 +19,38 @@
 
 @AutoValue
 public abstract class Method {
+  public enum Stream {
+    NONE,
+    CLIENT,
+    SERVER,
+    BIDI
```
Comment (summer-ji-eng) [inline src/main/java/com/google/api/generator/gapic/model/Method.java:26]: Can I have some context, what is `BIDI` :)

File: src/main/java/com/google/api/generator/gapic/protoparser/Parser.java [hunk 2]
```diff
@@ -54,16 +54,17 @@ public GapicParserException(String errorMessage) {
...
-        Service service =
-            Service.builder()
-                .setName(serviceDescriptor.getName())
-                .setPakkage(pakkage)
-                .setMethods(methods)
-                .build();
-        services.add(service);
-      }
+      List<Service> parsedServices =
+          fileDescriptor.getServices().stream()
```
Comment (summer-ji-eng) [inline src/main/java/com/google/api/generator/gapic/protoparser/Parser.java:58]: Very cool. Learned that this style is very similar with Javascript map

File: src/main/java/com/google/api/generator/gapic/protoparser/Parser.java [hunk 3]
```diff
@@ -78,25 +79,46 @@ public GapicParserException(String errorMessage) {
...
-                .setName(messageName)
-                .setFields(fields)
-                .build());
-      }
+      messages.putAll(parseMessages(fileDescriptor));
     }
     return messages;
   }
 
+  @VisibleForTesting
```
Comment (summer-ji-eng) [inline src/main/java/com/google/api/generator/gapic/protoparser/Parser.java:87]: Great. New annotation I learned from this PR.

File: src/main/java/com/google/api/generator/gapic/model/Method.java (hunk 1 above)
Comment (miraleung) [reply src/main/java/com/google/api/generator/gapic/model/Method.java:26]: **BiDi**rectional streaming, I think.

File: src/main/java/com/google/api/generator/gapic/protoparser/Parser.java (hunk 2 above)
Comment (miraleung) [reply src/main/java/com/google/api/generator/gapic/protoparser/Parser.java:58]: Great!

**Pull request merged**